Automattic/jetpack
# Jetpack Boost: Remove Header component from storybook

## Proposed changes:
Delete the storybook stories for the Header component in the Jetpack Boost plugin because it was breaking storybook tests. It's not a pure/reusable component - so it seems that it shouldn't be a part of storybook.

### Other information:



## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
N/A

## Testing instructions:
- Navigate to storybook
- Verify that the Header component is no longer present
